haproxy_agent.py

```diff
@@ -151,12 +152,17 @@ async def exapps_msg(path: str, headers: str, client_ip):
         async with EXAPP_CACHE_LOCK:
             exapp_record = EXAPP_CACHE.get(exapp_id_lower)
             if not exapp_record:
-                exapp_record = await nc_get_exapp(exapp_id_lower)
-                if not exapp_record:
-                    LOGGER.error("No such ExApp enabled: %s", exapp_id)
+                try:
+                    exapp_record = await nc_get_exapp(exapp_id_lower)
+                    if not exapp_record:
+                        LOGGER.error("No such ExApp enabled: %s", exapp_id)
+                        await record_ip_failure(client_ip_str)
+                        return reply.set_txn_var("not_found", 1)
+                    EXAPP_CACHE[exapp_id_lower] = exapp_record
+                except RequestFailException as e:
+                    LOGGER.exception("Failed to fetch ExApp metadata from Nextcloud", exc_info=e)
                     await record_ip_failure(client_ip_str)
```

Comment:
maybe delete this line? the client is not to blame that there are errors in the infrastructure - he does not need to be banned for this

Reply:
cool, I was a bit confused about this if bruteforce protection can be circumvented here